torch/_dynamo/variables/builder.py

```diff
@@ -799,8 +799,16 @@ def build_key_value(i, k, v):
             return SDPAParamsVariable.create(self.tx, value, self.source)
         elif isinstance(value, _EventBase):
             self.install_guards(GuardBuilder.ID_MATCH)
+            torch._dynamo.utils.store_user_object(value)
```

Comment:
Consider reflecting "weakref" in the name of this util function so it's clear that dynamo is not keeping them alive.